### PR TITLE
[Tracing] Add configuration consistency tests for DD_TRACE_HTTP_CLIENT_ERROR_STATUSES

### PR DESCRIPTION
**Pull request by zacharycmontoya**
## Motivation

We are systematically adding tests for tracing configurations, ensuring that they are implemented consistently across all languages.

## Changes

Adds test cases to the `DEFAULT` and `TRACING_CONFIG_NONDEFAULT` scenarios to cover the expected behaviors of the `DD_TRACE_HTTP_CLIENT_ERROR_STATUSES` configuration. All test cases fall into the already-defined `tracing_configuration_consistency` feature.

Note: This PR is stacked on top of https://github.com/DataDog/system-tests/pull/2980

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

**Review by robertomonteromiguel — APPROVED** [review]

File: tests/test_config_consistency.py [hunk 1]
```diff
@@ -71,3 +72,86 @@ def test_status_code_202(self):
...
+
+    def test_status_code_400(self):
+        assert self.r.status_code == 200
+        content = json.loads(self.r.text)
+        assert content["status_code"] == 400
+
+        interfaces.library.assert_trace_exists(self.r)
+        spans = [s for _, _, s in interfaces.library.get_spans(request=self.r, full_trace=True)]
+
+        client_span = _get_span_by_name(spans, "http.request")
```
Comment (mabdinur) [inline tests/test_config_consistency.py:93]: In python the name of the span will be `requests.request`. Can we instead get the span by resource name and span kind (ie match `GET /` and `client`). I think this value should be consistent across languages.

DD_TRACE_SPAN_ATTRIBUTE_SCHEMA is v0 by default in python.

File: tests/test_config_consistency.py [hunk 2]
```diff
@@ -71,3 +72,86 @@ def test_status_code_202(self):
...
+    def setup_status_code_400(self):
+        self.r = weblog.get("/make_distant_call", params={"url": "http://weblog:7777/status?code=400"})
+
+    def test_status_code_400(self):
+        assert self.r.status_code == 200
+        content = json.loads(self.r.text)
+        assert content["status_code"] == 400
+
+        interfaces.library.assert_trace_exists(self.r)
+        spans = [s for _, _, s in interfaces.library.get_spans(request=self.r, full_trace=True)]
```
Comment (mabdinur) [inline tests/test_config_consistency.py:91]: It took me way to long to find the `full_trace` parameter. In my tests the `get_spans` method was consistently returning only the root span so I started writing my own helper 😵‍💫.

File: tests/test_config_consistency.py (hunk 1 above)
Comment (mabdinur) [reply tests/test_config_consistency.py:93]: Ex: https://github.com/DataDog/system-tests/pull/3026/files#diff-4eba22e107ccb54fd5d8f0acdaf5c8eb131320d4cfbae71c4d664a4616eb1b15R102

Comment (zacharycmontoya) [reply tests/test_config_consistency.py:93]: Sounds good, will provide an update soon

Comment (zacharycmontoya) [reply tests/test_config_consistency.py]: Implemented in 6fb43882ec7a7490c3d1d75fac41258f51d33dab

**Review by mabdinur — APPROVED** [review]

**Pull request merged**